core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/CompactEntityComponent.scala

```diff
@@ -160,25 +216,41 @@ class CompactEntityQuery(driverComponent: DriverComponent) extends RawSqlQuery w
     *
     * `execution plan: batched insert (one statement, multiple rows)`
     */
-  def upsertReferences(fromId: Long, toIds: Set[Long]): ReadWriteAction[Int] = {
-    val insertValues: Iterable[SQLActionBuilder] = toIds.map { toId =>
-      sql"($fromId,$toId)"
-    }
-    val allInsertValues = reduceSqlActionsWithDelim(insertValues.toSeq, sql",")
+  def upsertReferences(fromId: Long, toIds: Set[Long]): ReadWriteAction[Int] =
+    upsertReferences(Set((fromId, toIds)))
 
-    val query = concatSqlActions(
-      sql"""insert into ENTITY_REFS(from_id, to_id)
+  /**
+    * Insert into ENTITY_REFS(from_id, to_id) values(...) on duplicate key update from_id=from_id
+    *
+    * Returns the number of rows upserted.
+    *
+    * `execution plan: batched insert (one statement, multiple rows)`
+    */
+  def upsertReferences(references: Set[(Long, Set[Long])]): ReadWriteAction[Int] =
```

Comment:
nit: use a case class: 
```
case class EntityReferences(fromId: Long, toIds: Set[Long])
```
And I think `Int` is correct, not `Long` but I am not certain.

Reply:
switched to using a case class.

I'm keeping this as `Long` to match the existing definitions on the entities table, e.g. https://github.com/broadinstitute/rawls/blob/f858c8ce71bd1e04463ab35064e57e39dfcd9414/core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/EntityComponent.scala#L51